src/main/java/bio/terra/testrunner/runner/TestScriptResult.java

```diff
@@ -2,41 +2,13 @@
 
 import bio.terra.testrunner.common.BasicStatistics;
 import bio.terra.testrunner.runner.config.TestScriptSpecification;
-import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import com.fasterxml.jackson.annotation.JsonIgnore;
 import java.util.List;
 import org.apache.commons.math3.stat.descriptive.DescriptiveStatistics;
 
 public class TestScriptResult {
   public List<UserJourneyResult> userJourneyResults;
-  public TestScriptResultSummary summary;
-
-  /**
-   * Summary statistics are pulled out into a separate inner class for easier summary reporting.
-   * This class does not include a reference to the full TestScriptSpecification or the list of
-   * UserJourneyResults.
-   */
-  @SuppressFBWarnings(
-      value = "URF_UNREAD_PUBLIC_OR_PROTECTED_FIELD",
-      justification = "This POJO class is used for easy serialization to JSON using Jackson.")
-  public static class TestScriptResultSummary {
-    public String testScriptName;
-    public String testScriptDescription;
-
-    public BasicStatistics elapsedTimeStatistics;
-
-    public int totalRun; // total number of user journey threads submitted to the thread pool
-    public int numCompleted; // number of user journey threads that completed
-    public int numExceptionsThrown; // number of user journey threads that threw exceptions
-
-    public boolean isFailure; // numCompleted < totalRun
-
-    public TestScriptResultSummary() {} // default constructor so Jackson can deserialize
-
-    private TestScriptResultSummary(String testScriptName, String testScriptDescription) {
-      this.testScriptName = testScriptName;
-      this.testScriptDescription = testScriptDescription;
-    }
-  }
+  @JsonIgnore public TestScriptResultSummary summary;
```

Comment:
I think you don't want `@JsonIgnore` here. RE your comment in the main PR, `SUMMARY_testRun.json` contains the `TestRunSummary`, which is different than the `TestScriptResultSummary` here.

Reply:
Replaced `testResultScripts` in `TestRunFullOutput` with List<Map<String, List<UserJourneyResult>> object. Effect is the same as using `@JsonIgnore`. So removed `@JsonIgnore` annotation.

Reply:
Sorry, I'm not following why you want to omit `TestScriptResultSummary` objects from the full JSON output. Don't you want information from those objects in the dashboard? Those objects hold all the statistics calculated about each set of test scripts (e.g. average time per user journey, 99 percentile time per user journey).

Reply:
The `TestRunFullOutput` already contains a list of `TestScriptResultSummary` results

`public List<TestScriptResultSummary> testScriptResultSummaries;`

and the summaries are populated by passing the `TestRunSummary` in the constructor method

```
  public TestRunFullOutput(
      TestConfiguration testConfiguration,
      List<Map<String, List<UserJourneyResult>>> testScriptResults,
      TestRunSummary testRunSummary,
      List<VersionScriptResult> versionScriptResults) {
...
   this.testScriptResultSummaries = testRunSummary.testScriptResultSummaries;
```

So I don't want to duplicate the summaries in testScriptResults again.

Let's have a zoom call if this isn't clear.

Reply:
Ah I see. Would it be easier to just remove the `TestRunFullOutput.testScriptResultSummaries` property instead?

Reply:
Gotcha. Agreed we should do whatever makes sense for the dashboard in the `TestRunFullOutput` class, since that's its main purpose.

Not sure I'm following the `List<Map<String, List<UserJourneyResult>>>` though -- aren't you just creating a list of maps with a single identical key `userJourneyResults`? Isn't that just a list of `List`s -- why do you need the `Map`?

It feels like we're missing a mapping here though. If you're looking at a `TestScriptResultSummary`, there's no way to link that back to the `UserJourneyResult`s for that same `TestScript`. I think we didn't have this problem before because the `TestScriptResultSummary` and the `List<UserJourneyResult>` are in the same `TestScriptResult` object. Now that you'll separate them, do we need a test script id or something to be able to link them? What do you think?

Reply:
Yes we can link back the `UserJourneyResult`s to the corresponding `TestScriptResultSummary`. I added the following comments to the `TestRunFullOutput`. Using the position argument would be sufficient.

```
  // The testScriptResults and testScriptResultSummaries must have identical size.
  // For each element in testScriptResults, we can derive the testScriptName and
  // testScriptDescription properties from the corresponding positional element in
  // testScriptResultSummaries.
  public List<Map<String, List<UserJourneyResult>>> testScriptResults;
  public List<TestScriptResultSummary> testScriptResultSummaries;
```

Reply:
Another more flexible is to have an explicit `unnestUserJourneyResults` method that can shape the object we desire for automated reporting. I just implemented this method and checked in the code,

Reply:
Regarding your other comments. Quote

```
Not sure I'm following the List<Map<String, List<UserJourneyResult>>> though -- aren't you just creating a list of maps with a single identical key userJourneyResults? Isn't that just a list of Lists -- why do you need the Map?
```

List of **maps** is easier to deal with than list of **lists**. For list of maps, I can unnest the top-level list first and access the list of user journeys by the key name. For list of lists, I ran into trouble using indices to access the result. ORM (SQLAlchemy) works better but I haven't tried out yet.

Reply:
Makes sense. Perhaps include that as a comment? I think all these troubles you're running into creating a BQ easily-queryable schema are good information for the next person who has to modify this.